### PR TITLE
fix(ocsp): disable periodic refresh when listener or stapling are disabled

### PR DESCRIPTION
**Pull request by thalesmg**
Fixes https://emqx.atlassian.net/browse/EMQX-9773

# Targeting `release-50`

### <samp>🤖 Generated by Copilot at 3f3f63c</samp>

This pull request improves the OCSP stapling feature of EMQ X by adding logic to handle the case when OCSP stapling is disabled for a listener. It also enhances the test suite for the `emqx_ocsp_cache` module. The main files affected are `emqx_listeners.erl`, `emqx_ocsp_cache.erl`, and `emqx_ocsp_cache_SUITE.erl`.

## PR Checklist
Please convert it to a draft if any of the following conditions are not met. Reviewers may skip over until all the items are checked:

- [x] Added tests for the changes
- [x] Changed lines covered in coverage report
- [na] Change log has been added to `changes/{ce,ee}/(feat|perf|fix)-<PR-id>.en.md` files
- [x] For internal contributor: there is a jira ticket to track this change
- [na] If there should be document changes, a PR to emqx-docs.git is sent, or a jira ticket is created to follow up
- [na] Schema changes are backward compatible

## Checklist for CI (.github/workflows) changes

- [na] If changed package build workflow, pass [this action](https://github.com/emqx/emqx/actions/workflows/build_packages.yaml) (manual trigger)
- [na] Change log has been added to `changes/` dir for user-facing artifacts update

### CODE REVIEW

File: apps/emqx/test/emqx_ocsp_cache_SUITE.erl [hunk 1]
```diff
@@ -44,6 +44,8 @@ groups() ->
     ].
 
 init_per_suite(Config) ->
+    emqx_common_test_helpers:clear_screen(),
```
Comment (savonarola) [inline apps/emqx/test/emqx_ocsp_cache_SUITE.erl:47]: Was forgotten to remove?

File: apps/emqx/test/emqx_ocsp_cache_SUITE.erl [hunk 2]
```diff
@@ -739,6 +748,71 @@ do_t_update_listener(Config) ->
         )
     ),
     assert_http_get(1, 5_000),
+
+    %% Disable OCSP Stapling; the periodic refreshes should stop
+    RefreshInterval = emqx_config:get([listeners, ssl, default, ssl_options, ocsp, refresh_interval]),
+    ct:print("interval >>>>>>>>>>>> ~p", [RefreshInterval]),
```
Comment (savonarola) [inline apps/emqx/test/emqx_ocsp_cache_SUITE.erl:754]: Remove?

File: apps/emqx/test/emqx_ocsp_cache_SUITE.erl (hunk 1 above)
Comment (thalesmg) [reply apps/emqx/test/emqx_ocsp_cache_SUITE.erl]: :see_no_evil: 
:heavy_check_mark:

File: apps/emqx/test/emqx_ocsp_cache_SUITE.erl (hunk 2 above)
Comment (thalesmg) [reply apps/emqx/test/emqx_ocsp_cache_SUITE.erl]: :see_no_evil: 
:heavy_check_mark:

File: apps/emqx/src/emqx_listeners.erl [hunk 3]
```diff
@@ -460,10 +462,18 @@ post_config_update([listeners, Type, Name], {action, _Action, _}, NewConf, OldCo
     #{enabled := NewEnabled} = NewConf,
     #{enabled := OldEnabled} = OldConf,
     case {NewEnabled, OldEnabled} of
-        {true, true} -> restart_listener(Type, Name, {OldConf, NewConf});
-        {true, false} -> start_listener(Type, Name, NewConf);
-        {false, true} -> stop_listener(Type, Name, OldConf);
-        {false, false} -> stop_listener(Type, Name, OldConf)
+        {true, true} ->
+            maybe_unregister_ocsp_stapling_refresh(Type, Name, NewConf),
```
[the remark this answers is not in the document]
Comment (thalesmg) [reply apps/emqx/src/emqx_listeners.erl]: :heavy_check_mark:

**Review by savonarola — APPROVED** [review]

**Pull request merged**